Application/ControlMsrE2/ControlMsrE2UI.c

```diff
@@ -0,0 +1,241 @@
+      case CHAR_BACKSPACE:
+        if (Pos > 0) {
+          --Pos;
+          gST->ConOut->SetCursorPosition (gST->ConOut, StartColumn + Pos, StartRow);
+          gST->ConOut->OutputString (gST->ConOut, L" ");
+          gST->ConOut->SetCursorPosition (gST->ConOut, StartColumn + Pos, StartRow);
+        }
+        break;
+
+      case 0x1B:
+        if (Pos > 0) {
+          Pos = 0;
+          gST->ConOut->SetCursorPosition (gST->ConOut, StartColumn + Pos, StartRow);
+          for (Index = 1; Index < Length; ++Index) {
+            gST->ConOut->OutputString (gST->ConOut, L" ");
+          }
+          gST->ConOut->SetCursorPosition (gST->ConOut, StartColumn + Pos, StartRow);
+        } else {
+          Buffer[Pos] = 0;
+          return Pos;
+        }
+        break;
+
+      case CHAR_CARRIAGE_RETURN:
+      case CHAR_LINEFEED:
+        Buffer[Pos] = 0;
+        gST->ConOut->EnableCursor (gST->ConOut, FALSE);
+        return Pos;
+
+      default:
+        Buffer[Pos] = Key.UnicodeChar;
+        ++Pos;
+        Output[0] = Key.UnicodeChar;
+        gST->ConOut->OutputString (gST->ConOut, Output);
+
+        if (Pos >= Length - 1) {
+          Buffer[Pos] = 0;
+          return Pos;
+        }
+        break;
+    }
+  } while (TRUE);
+
+  return 0;
+}
+
+CHAR16
+ReadAnyKey (
+  VOID
+  )
+{
+  CHAR16 Keys[2];
+  ReadLine (Keys, 2);
+  return Keys[0];
+}
+
+BOOLEAN
+ReadYN (
+  VOID
+  )
+{
+  CHAR16 Keys[2];
+  do {
+    ReadLine (Keys, 2);
+  } while (OcStriStr (L"yn", Keys) == 0);
+
+  return Keys[0] == 'y' || Keys[0] == 'Y';
+}
+
+#define TOKENLENGTH 32
+
+EFI_STATUS
+InterpretArguments (
+  VOID
+  )
+{
+  UINTN       Argc;
+  CHAR16      **Argv;
+  CHAR16      *Parameter;
+  UINTN       ParameterCount;
+  CHAR16      *Token;
+  UINT16      TokenIndex;
+  UINT32      Index;
+
+  Flags = 0;
+
+  if (EFI_ERROR (GetArguments (&Argc, &Argv))) {
+    Argc = 1;
+  }
+
+  ParameterCount = 0;
+
+  for (Index = 1; Index < Argc; ++Index) {
+    Token = AllocatePool (StrSize (Argv[Index]));
+
+    if (Token != NULL) {
+      StrCpyS (Token, StrLen (Argv[Index]) + 1, Argv[Index]);
+
+      TokenIndex = 0;
+
+      while (Argv[Index][TokenIndex] != '\0') {
+        while (Argv[Index][TokenIndex] == ' ') {
+          ++TokenIndex;
+        }
+
+        if (Argv[Index][TokenIndex] != '\0') {
+          Parameter = &Token[TokenIndex];
+
+          while (Token[TokenIndex] != '\0' && Token[TokenIndex] != ' ') {
+            ++TokenIndex;
+          }
+          Token[TokenIndex] = '\0';
+
+          if (!StrCmp (Parameter, L"lock")) {
+            Flags |= ARG_LOCK;
+            ++ParameterCount;
+          } else if (!StrCmp (Parameter, L"unlock")) {
+            Flags |= ARG_UNLOCK;
+            ++ParameterCount;
+          } else if (!StrCmp (Parameter, L"interactive")) {
+            Flags |= ARG_INTERACTIVE;
+            ++ParameterCount;
+          } else {
+            Print (L"Ignoring unknown command line argument: %s\n", Parameter);
```

Comment:
I think most tools error when the arguments passed make no sense, especially in UEFI I believe we should anyway. Please also refer to the "options mutually exclusive" comments regarding this.